### PR TITLE
[WIP] Refactor/major change with k1st

### PR DESCRIPTION
**Pull request by vuonghoainam**
- [x] Use just 1 core term: h1st `model` in `h1st.core.model`
- [x] Simple code based on Core `Model`
- [x] Remove redundant code for `model`/ `modeler` and other types of model
- [x] Temporary disable `load()` and `save()` of `model` due to redesigned `ModelRepository`

- [x] Fix import
- [ ] ...

### CODE REVIEW

File: user/example-models/simple_model.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# USER: Other framework (K1st) and service layer
+
+# WHAT:
+# Simple, understandable in less than 5 seconds for the term
+# Stressing on Knowledge
+
+import os
+import sys
+sys.path.append(os.path.join(os.path.dirname(__file__),'../../'))
```
Comment (ctn) [inline user/example-models/simple_model.py:9]: This should not be necessary in the actual code base, once you have the proper package.

File: h1st/model/model.py [hunk 2]
```diff
@@ -45,6 +44,14 @@ class MyModel(h1st.model.Model):
     """
 
     ## TODO: Need a better naming and the definition of the property
+    @property
+    def model_class(self) -> Any:
+        return getattr(self, "__model_class", None)
+
+    @model_class.setter
+    def model_class(self, value):
```
Comment (ctn) [inline h1st/model/model.py:52]: Why do we need to expose `model_class`?

File: h1st/model/model.py [hunk 3]
```diff
@@ -95,3 +109,49 @@ def process(self, input_data: Dict) -> Dict:
         """
         # not raise NotImplementedError so the initial model created by integrator will just work
         return input_data
+
+    def train(self, data: Dict[str, Any] = None) -> None:
+        """
+        Implement logic to create the corresponding MLModel, including both training and evaluation.
```
Comment (ctn) [inline h1st/model/model.py:115]: Explain what we are doing here, which is to call the appropriate `train` or `fit` on the `base_model`, using the prepared training data.

File: user/example-apps/koracle/models/oracle/oracle.py [hunk 4]
```diff
@@ -2,13 +2,10 @@
 
 import pandas as pd
 
-import __init__
-from h1st.model.rule_based_model import RuleBasedModel
+from h1st.model.knowledge_model import RuleBasedModel
```
Comment (ctn) [inline user/example-apps/koracle/models/oracle/oracle.py:5]: Shouldn’t this be removed entirely? It belongs in `k1st`

**Review by ctn — CHANGES_REQUESTED** [review]
Review comments and after that we are probably good to merge.

File: user/example-apps/koracle/models/oracle/oracle.py (hunk 4 above)
Comment (vuonghoainam) [reply user/example-apps/koracle/models/oracle/oracle.py]: I removed this one.
This has been moved to k1st

File: user/example-models/simple_model.py (hunk 1 above)
Comment (vuonghoainam) [reply user/example-models/simple_model.py]: I have removed this part completely

File: h1st/model/model.py (hunk 3 above)
Comment (ctn) [reply h1st/model/model.py:70]: Moving fast.

**Review by ctn — APPROVED** [review]
Moving fast so lgtm.

**Comment by ctn** [comment]
Merging and moving ahead @vuonghoainam

**Pull request merged**